src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_utils.py

```diff
@@ -0,0 +1,40 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+# pylint: disable=no-self-use,too-many-arguments,no-member,line-too-long
+
+from os.path import exists, join
+from OpenSSL import crypto
+
+
+def create_self_signed_certificate(device_id, valid_days, cert_output_dir):
```

Comment:
Haven't heard of any plans.
The major scenario we want to enable here is that user can associate certificate thumbprints with their devices. How they manage their certificates will be up to themselves. Key vault is certainly a recommended solution. For now, we just leave the choices to users.
The self-signed certificate functionality is intended for developers. It would help developer to quickly try the certificate authentication approach, which is recommended for better security.

---
In reply to: [87032196](https://github.com/Azure/azure-cli/pull/1263#discussion_r87032196) [](ancestors = 87032196)